src/Validation/Rules/Ulid.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Validation\Rules;
+
+use Attribute;
+use Tempest\Validation\Rule;
+
+#[Attribute]
+final readonly class Ulid implements Rule
+{
+    public function isValid(mixed $value): bool
+    {
+        return preg_match('/^[0-9A-HJKMNP-TV-Z]{26}$/', $value) === 1;
```

Comment:
return preg_match('/^[0-9A-HJKMNP-TV-Z]{26}$/', strtoupper($value)) === 1;

Reply:
@didi955 - Why opt for `strtoupper` over including those characters in the expression?